src/main/java/org/minions/devfund/damianvp/moviedatabase/MovieDatabase.java

```diff
@@ -0,0 +1,150 @@
+                }
+            }
+            this.movieList.add(movie);
+        }
+    }
+
+    /**
+     * Method to check if movie already exist in class.
+     * @param name String type.
+     * @return true: if movie exist.
+     *         false: if movie does not exist.
+     */
+    public boolean containsMovie(final String name) {
+        return this.movieList.stream().anyMatch(o -> o.getName().equals(name));
+    }
+
+    /**
+     * Method to check if actor already exist in class.
+     * @param name String type.
+     * @return true: if actor exist.
+     *         false: if actor does not exist.
+     */
+    public boolean containsActor(final String name) {
+        return this.actorList.stream().anyMatch(o -> o.getName().equals(name));
+    }
+
+    /**
+     * Method to get the Actor object.
+     * @param name String type.
+     * @return Actor object.
+     */
+    public Actor getActorObject(final String name) {
+        return this.actorList.stream().findAny().filter(o -> o.getName().equals(name)).orElse(new Actor(name));
+    }
+
+    /**
+     * Method to get the Movie object form the list by movie name.
+     * @param name String type.
+     * @return Movie object from list: if movie already exists.
+     *         new Movie object: if movie does not exists.
+     */
+    public Movie getMovieObject(final String name) {
+        Movie ob = this.movieList.stream().findAny().filter(o -> o.getName().equals(name)).orElse(new Movie(name));
+        return ob;
+    }
+
+    /**
+     * Method to add a rating to a specific movie.
+     * @param name String type, name of movie.
+     * @param rating double type, rating to set.
+     */
+    void addRating(final String name, double rating) {
+        for (int i = 0; i < this.movieList.size(); i++) {
+            if (this.movieList.get(i).getName().equals(name)) {
+                this.movieList.get(i).setRating(rating);
+                return;
+            }
+        }
+    }
+
+    /**
+     * Method to update the rating of a specific movie.
+     * @param name String type, name of movie.
+     * @param newRating double type, rating to update.
+     */
+    void updateRating(final String name, double newRating) {
+        addRating(name, newRating);
+    }
+
+    /**
+     * Method to get the nest actor of class.
+     * @return String type, name of the best actor.
+     */
+    String getBestActor() {
+        Comparator<Actor> comparator = Comparator.comparing(Actor::getRatingAverage);
+        String name = this.actorList.stream().max(comparator).get().getName();
+        Actor actor = getActorObject(name);
+        return actor.getName();
+    }
+
+    /**
+     * Method to get the best movie of the class.
+     * @return String type, name of best movie.
+     */
+    String getBestMovie() {
+        Comparator<Movie> comparator = Comparator.comparing(Movie::getRating);
+        return this.movieList.stream().max(comparator).get().getName();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)